File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/bids/dto/BidHistoryResponse.java
package com.portfolio.auctionmarket.domain.bids.dto;
import com.fasterxml.jackson.annotation.JsonUnwrapped;
import com.portfolio.auctionmarket.domain.products.dto.ProductAndAuctionResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BidHistoryResponse {

    @JsonUnwrapped
    private ProductAndAuctionResponse response;
    private Long myMaxBidPrice;

}

File: AuctionMarket/auction-domain/src/main/kotlin/org/example/auctioncommon/domain/bid/dto/BidHistoryResponse.kt
package org.example.auctioncommon.domain.bid.dto

import com.fasterxml.jackson.annotation.JsonUnwrapped
import org.example.auctioncommon.domain.product.dto.ProductAndAuctionResponse

class BidHistoryResponse(
    @field:JsonUnwrapped
    private val response: ProductAndAuctionResponse? = null,
    private val myMaxBidPrice: Long? = null
) {

}
